vql/darwin/xattr.go

```diff
@@ -0,0 +1,69 @@
+//go:build darwin
+// +build darwin
+
+package darwin
+
+import (
+	"context"
+
+	"github.com/Velocidex/ordereddict"
+	"github.com/ivaxer/go-xattr"
+	"www.velocidex.com/golang/velociraptor/acls"
+	vql_subsystem "www.velocidex.com/golang/velociraptor/vql"
+	"www.velocidex.com/golang/vfilter"
+	"www.velocidex.com/golang/vfilter/arg_parser"
+)
+
+type XAttrArgs struct {
+	Filename  string `vfilter:"required,field=filename,doc=Filename to inspect."`
+	Attribute string `vfilter:"required,field=attribute,doc=Attribute to collect."`
+}
+
+type XAttrPlugin struct{}
+
+func (self XAttrPlugin) Call(
+	ctx context.Context,
+	scope vfilter.Scope,
+	args *ordereddict.Dict) <-chan vfilter.Row {
+	output_chan := make(chan vfilter.Row)
+
+	go func() {
+		defer close(output_chan)
+		defer vql_subsystem.CheckForPanic(scope, "xattr")
+
+		arg := &XAttrArgs{}
+		err := arg_parser.ExtractArgsWithContext(ctx, scope, args, arg)
+		if err != nil {
+			scope.Log("xattr: %s", err.Error())
+			return
+		}
+
+		var data []byte
+		data, err = xattr.Get(arg.Filename, arg.Attribute)
```

Comment:
We need an accessor here too - you can get the filename with GetUnderlyingAPIFilename()

https://github.com/Velocidex/velociraptor/blob/58e92c447179c6dc7a5a09be7fe58cf0ef39c8c4/accessors/api.go#L331